src/Connection.php

```diff
@@ -141,6 +139,36 @@ public function getDatabaseName()
         return $this->getMongoDB()->getDatabaseName();
     }
 
+    public function enableQueryLog()
+    {
+        parent::enableQueryLog();
+
+        $this->commandSubscriber = new CommandSubscriber($this);
+        $this->connection->addSubscriber($this->commandSubscriber);
+    }
+
+    public function disableQueryLog()
+    {
+        parent::disableQueryLog(); // TODO: Change the autogenerated stub
+
+        $this->connection->removeSubscriber($this->commandSubscriber);
```

Comment:
Needs a safe guard in case no subscriber is registered

Reply:
Added a condition to never add or remove the subscriber twice.